src/OpenSearch.Net/Transport/PostData.cs

```diff
@@ -85,8 +85,8 @@ public abstract class PostData
 
 		public static PostData Bytes(byte[] bytes) => new PostData<object>(bytes);
 
-#if NETSTANDARD2_1
-		public static PostData ReadOnlyMemory(ReadOnlyMemory<byte> bytes) => new PostData<object>(bytes);
+#if NETSTANDARD2_1 || NET6_0_OR_GREATER
+		public static PostData ReadOnlyMemory(ReadOnlyMemory<byte> bytes) => new PostData<object>(bytes.ToArray());
```

Comment:
Please remove the `.ToArray()`, this should be using the constructor here: https://github.com/opensearch-project/opensearch-net/blob/main/src/OpenSearch.Net/Transport/PostData.cs#L149

You'll just need to update the `#if`s further down.

Reply:
I think I understood this comment, it is in this commit bcbd1a2161d263ecb010daf522896965ba39531f